File: data_validators/url_sanitizer.py
endpoint = "http://geco.deib.polimi.it/cov2k/api/variants/"
args = {
    "nextsrain-id": "20A/S:98F",
    "pangolin-id": "21C (Epsilon)"
}

if __name__ == '__main__':
    from urllib.parse import urljoin, urlencode

    url = urljoin(endpoint, urlencode(args, doseq=True))
    print(url, type(url))

# encodes array as of type:form explode:True

File: data_validators/url_sanitizer.js
const endpoint = new URL("http://geco.deib.polimi.it/cov2k/api/variants/");
var args = new URLSearchParams({
    "nextsrain-id": "20A/S:98F",
    "pangolin-id": "21C (Epsilon)",
    "effects": [1, 2, 4]
});

args.forEach((value, key) => {
  endpoint.searchParams.append(key, value);
});
console.log(endpoint.toString())

// encodes arrays as comma separated list of values, but the comma is encoded too.